module/zfs/vdev_indirect.c

```diff
@@ -870,8 +870,8 @@ spa_start_indirect_condensing_thread(spa_t *spa)
 
 /*
  * Gets the obsolete spacemap object from the vdev's ZAP.
- * Returns the spacemap object, or 0 if it wasn't in the ZAP or the ZAP doesn't
- * exist yet.
+ * Returns the spacemap object, or 0 if it wasn't in the ZAP,
+ * the ZAP doesn't exist yet, or the ZAP is damaged.
```

Comment:
I don't think that this semantic change is safe.  Have you examined all callers to ensure that they can safely ignore the presence of this object?